### ISSUE TITLE
Questions regarding use of riscv-dv

### ISSUE THREAD

**Issue by MikeOpenHWGroup** [openhwgroup/cve2#159]
### Bug Description

This Issue is motivated by pull-request #150, which includes update to [gen_csr_test.py](https://github.com/openhwgroup/cve2/blob/d2a1be82ef5d68a179cef1972a2f304cee459ce8/vendor/google_riscv-dv/scripts/gen_csr_test.py#L1), a python script included in the Chips Alliance [riscv-dv](https://github.com/chipsalliance/riscv-dv) instruction stream generator.

The Ibex repository from which CVE2 is forked included a specific version of riscv-dv as a component of its Ibex verification environment.  In OpenHW we do not use this environment, as we have adopted [core-v-verif](https://github.com/openhwgroup/core-v-verif/tree/cv32e20/dev/cv32e20) for CVE2.  Pull-request #21 removed the `dv` directory from the CVE2 repo.

Given the above, I have two questions:
1. Should we delete the [vendor/google_riscv-dv](https://github.com/openhwgroup/cve2/tree/main/vendor/google_riscv-dv) directory from this repository?
2. Where should the CVE2-specific version of `gen_csr_test.py` go?

My recommendations:
1. Yes, we should delete `vendor/google_riscv-dv`.  Now that `dv` has been deleted, we do not use it anywhere.
2. My suggestion is to place `gen_csr_test.py` in [cv32e20/env/corev-dv](https://github.com/openhwgroup/core-v-verif/tree/cv32e20/dev/cv32e20/env/corev-dv).

Follow-on question:
- Should the CV32E20-specific version of `gen_csr_test.py` be implemented as an extension of the riscv-dv script, or should we simply copy-and-edit the riscv-dv code to create the CV32E20-specific version of `gen_csr_test.py`?

**Comment by emgens** [comment]
I agree with deleting ` vendor/google_riscv-dv `;  It makes far more sense in my mind to have this script under core-v-verif.  

I'm not 100% certain that [cv32e20/env/corev-dv](https://github.com/openhwgroup/core-v-verif/tree/cv32e20/dev/cv32e20/env/corev-dv) is the correct place for it.  It is basically just generating a test, so it might make more sense to have it checked into the relevant [cv32e20/tests/programs/custom/](https://github.com/openhwgroup/core-v-verif/tree/cv32e20/dev/cv32e20/tests/programs/custom/) sub-directory (in pull-request https://github.com/openhwgroup/core-v-verif/pull/2220, I called that riscv_csr, but I'm open to changing the name.)  

And as much as I'd like this to some how be an extension of the existing ` gen_csr_test.py`, the structure of the code doesn't lend itself to that.  So I think this needs to be copy and edit.
